src/secret-storage.ts

```diff
@@ -357,21 +363,33 @@ export class ServerSideSecretStorageImpl implements ServerSideSecretStorage {
     }
 
     /**
-     * Set the default key ID for encrypting secrets.
-     *
-     * @param keyId - The new default key ID
+     * Implementation of {@link ServerSideSecretStorage#setDefaultKeyId}.
      */
-    public setDefaultKeyId(keyId: string): Promise<void> {
+    public setDefaultKeyId(keyId: string | null): Promise<void> {
         return new Promise<void>((resolve, reject) => {
             const listener = (ev: MatrixEvent): void => {
-                if (ev.getType() === "m.secret_storage.default_key" && ev.getContent().key === keyId) {
+                if (ev.getType() !== "m.secret_storage.default_key") {
+                    //  Different account data item
+                    return;
+                }
+
+                const content = ev.getContent();
+                // If keyId === null, the content should be an empty object.
+                // Otherwise, keyId should be returned in the content object.
+                const isSameKey = keyId === null ? !Object.keys(content).length : content.key === keyId;
```

Comment:
Slight clarifications. Let's avoid the implicit cast from number to boolean.

```suggestion
                // If keyId === null, the content should be an empty object.
                // Otherwise, the `key` in the content object should match keyId.
                const content = ev.getContent();
                const isSameKey = keyId === null ? Object.keys(content).length === 0 : content.key === keyId;
```